### ISSUE TITLE
[PlanningAlerts/production] ActiveRecord::RecordNotUnique: Mysql2::Error: Duplicate entry '[email]' for key 'index_users_on_email'

### ISSUE THREAD

**Issue by mlandauer** [openaustralia/planningalerts#1632]
## Backtrace

    line 26 of [PROJECT_ROOT]/app/controllers/users/registrations_controller.rb: create

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/37075/faults/91906356)

**Comment by mlandauer** [comment]
It's still happening. Sigh.

**Comment by mlandauer** [comment]
I think I know what's going on now. mysql string comparison is case insensitive but the default devise validations for the email are explicitly set to be case sensitive (I think so that they're backwards compatible with previous devise versions). Anyway it explains why the validations don't fail while mysql borks. Also in honeybadger I can see that the two most recent times this problem has occurred are due to case differences (though sometimes hard to spot).

**Comment by mlandauer** [comment]
Not quite true - devise actually does have an option to make the email comparison case insensitive (which is on in our case) but it does this not through the validation. The problem in our case is that we created a bunch of users with email addresses directly not through devise where their email addresses had upper case letters in it which devise expects not to happen. So, we need to change those records. This is similar to what we did for whitespace.
